src/rp/util/logic.clj

```diff
@@ -0,0 +1,7 @@
+(ns rp.util.logic)
+
+(defn xor
+  [p q]
```

Comment:
Cool! Here's a variation that would do less work when there are many truthy items:

```clojure
(defn xor
  ([] nil)
  ([x] x)
  ([x & next]
   (let [xs (cons x next)
         truthies (take 2 (filter identity xs))]
     (when (= (count truthies) 1)
       (first truthies)))))
```